src/main/java/ch/bfh/ti/i4mi/mag/Config.java

```diff
@@ -351,35 +386,36 @@ public OutPayloadLoggerInterceptor soapRequestLogger() {
     public InPayloadLoggerInterceptor soapResponseLogger() {
         return new InPayloadLoggerInterceptor("./logs/[date('yyyyMMdd-HH00')]/[sequenceId]-soap-response.txt");
     }
-    
+
     // ---------------------------------------------
     // Other beans used
     // ---------------------------------------------
-    
+
     @Bean
     public SchemeMapper getSchemeMapper() {
-    	return new SchemeMapper();
+        return new SchemeMapper();
     }
-    
+
     @Bean
     public PatientReferenceCreator getPatientReferenceCreator() {
-    	return new PatientReferenceCreator();
+        return new PatientReferenceCreator();
     }
 
     @Bean
-    public MagCapabilityStatementProvider serverConformanceProvider(
-            final RestfulServer fhirServer,
-            @Value("${mag.baseurl}") final String baseUrl) {
-        return new MagCapabilityStatementProvider(fhirServer, baseUrl);
+    public IServerConformanceProvider<IBaseConformance> serverConformanceProvider(FhirContext fhirContext,
+                                                                                  RestfulServerConfiguration theServerConfiguration) {
+        return new ServerCapabilityStatementProvider(fhirContext, theServerConfiguration);
     }
 
     // use to fix https://github.com/i4mi/MobileAccessGateway/issues/56, however we have the CapabilityStatement not filled out anymore
-	@Bean
+    @SuppressWarnings("unchecked")
```

Comment:
`new ArrayList<>()` is working fine, why are you adding the type at first, removing the chevrons in the next line and adding the SuppressWarnings annotation?